dds/DCPS/Time_Helper.h

```diff
@@ -23,14 +24,25 @@ OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
 namespace OpenDDS {
 namespace DCPS {
 
-ACE_INLINE OpenDDS_Dcps_Export
-ACE_Time_Value time_to_time_value(const DDS::Time_t& t);
+template <typename TimeT>
+ACE_Time_Value time_to_time_value(const TimeT& t)
+{
+  ACE_Time_Value tv(t.sec, t.nanosec / 1000);
+  return tv;
+}
+
+template <typename IdlType>
+void time_value_to_time(IdlType& t, const ACE_Time_Value& tv)
+{
+  t.sec = ACE_Utils::truncate_cast<CORBA::Long>(tv.sec());
+  t.nanosec = ACE_Utils::truncate_cast<CORBA::ULong>(tv.usec() * 1000);
+}
 
 ACE_INLINE OpenDDS_Dcps_Export
 DDS::Time_t time_value_to_time(const ACE_Time_Value& tv);
 
 ACE_INLINE OpenDDS_Dcps_Export
-MonotonicTime_t time_value_to_monotonic_time(const ACE_Time_Value& tv);
+MonotonicTime_t time_value_to_time(const ACE_Time_Value_T<ACE_Monotonic_Time_Policy>& tv);
```

Comment:
```suggestion
MonotonicTime_t time_value_to_time(const ACE_Time_Value_T<MonotonicClock>& tv);
```

I don't know if this would actually cause a problem, but the clock type should probably be consistent with `MonotonicTimePoint`.

Reply:
When I tried this change it failed to compile https://github.com/OpenDDS/OpenDDS/actions/runs/9417476930/job/25942737566  So for now it's reverted.

I agree with the idea, we would just need to work out the declaration order of the relevant pieces so that it doesn't lead to a cycle of `#include`s.

Reply:
Oh, yea, I guess `TimeTypes.h` needs this header... The clock definitions probably would need to go in their own header then?